### PR TITLE
lxc: update to 5.0.1

### PR DESCRIPTION
**Pull request by graysky2**
Bump to latest and update Makefile to use meson which is upstream's standard.
Deleted unneeded 010-Remove-distro-check.patch (reference to configure).

Build system: x86_64
Build-tested: bcm2711/RPi4B
Run-tested: bcm2711/RPi4B

Signed-off-by: John Audia <therealgraysky@proton.me>

Maintainers: @ratkaj

### CODE REVIEW

**Comment by graysky2** [comment]
This closes issue #18989

**Comment by graysky2** [comment]
My test build did not include all of the lxc tools.  I am not sure why lxc-monitord is failing.  Any suggestions are appreciated.

**Comment by graysky2** [comment]
Is this something we can fix in our `Makefile` or is this something upstream?

I see the expected file, `lxc-monitord` is getting put in the the libexe path:
```
Installing src/lxc/cmd/lxc-monitord to /scratch/union/build_dir/target-aarch64_cortex-a72_musl/lxc-5.0.0/ipkg-install/usr/libexec/lxc
```

But later on, `install` is directed to another path:
```
install -m0755 /scratch/union/build_dir/target-aarch64_cortex-a72_musl/lxc-5.0.0/ipkg-install"/usr/lib/lxc"/lxc-monitord /scratch/union/build_dir/target-aarch64_cortex-a72_musl/lxc-5.0.0/ipkg-aarch64_cortex-a72/lxc-monitord"/usr/lib/lxc"/
install: cannot stat '/scratch/union/build_dir/target-aarch64_cortex-a72_musl/lxc-5.0.0/ipkg-install/usr/lib/lxc/lxc-monitord': No such file or directory
```

File: utils/lxc/Makefile [hunk 1]
```diff
@@ -25,6 +25,19 @@ PKG_BUILD_PARALLEL:=1
 PKG_USE_MIPS16:=0
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/meson.mk
+
+TARGET_CFLAGS += -fPIC
+TARGET_LDFLAGS += -fPIC
```
Comment (neheb) [inline utils/lxc/Makefile:31]: meson has -Db_pic=true

Comment (neheb) [reply utils/lxc/Makefile:31]: And actually you probably mean to use -Db_pie.

File: utils/lxc/Makefile [hunk 2]
```diff
@@ -25,6 +25,19 @@ PKG_BUILD_PARALLEL:=1
...
+
+MESON_ARGS += \
+	-Dwerror=false \
+	-Dinit-script=sysvinit \
+	-Dman=false \
+	-Dapparmor=false \
+	-Dselinux=false \
+	-Dseccomp=$(if $(CONFIG_LXC_SECCOMP),true,false) \
```
Comment (neheb) [inline utils/lxc/Makefile:39]: this should probably be added to PKG_CONFIG_DEPENDS.

**Comment by neheb** [comment]
```
PKG_FIXUP:=autoreconf
PKG_INSTALL:=1
PKG_BUILD_PARALLEL:=1
```

should all be removed. the last two are default and the first is useless.

**Comment by graysky2** [comment]
@ratkaj - do you approve of this?

File: utils/lxc/Makefile (hunk 1 above)
Comment (BKPepe) [reply utils/lxc/Makefile:28]: I see this discussion is marked as resolved, but I dont see any change related to it?

**Comment by graysky2** [comment]
Is the maintainer of this package active?

**Comment by neheb** [comment]
No. Anyway, no idea why

```
TARGET_CFLAGS += -fPIC
TARGET_LDFLAGS += -fPIC
```

is insisted on. b_pic is default on.

**Comment by graysky2** [comment]
@neheb - I removed the two lines:
```
TARGET_CFLAGS += -fPIC
TARGET_LDFLAGS += -fPIC
```

and added `-Db_pie=true` to `MESON_ARGS` and it test built fine for bcm27xx.  Let's see if the build tests for others pass.

**Comment by graysky2** [comment]
@neheb -

https://github.com/openwrt/packages/runs/8246554550?check_suite_focus=true#step:5:4137

Seems to want to have `-fPIC` ... do you have any ideas?  Bug in upstream code?  Revert to defining `-fPIC` for now?

**Comment by AreYouLoco** [comment]
Is there anything I could do to help out testing lxc 5.0? I really need support for vlans added in that version

**Comment by graysky2** [comment]
@AreYouLoco - just pull this PR and build.  I have been using it since I opened the PR.

**Comment by AreYouLoco** [comment]
I'd rather wait until its upstreamed. I am totally not familiar with openwrt build system

**Comment by neheb** [comment]
I'll look into this failure. Sounds like an upstream bug.

**Comment by eli-schwartz** [comment]
Why does an ***executable*** want -fPIC? Meson's `b_pic` option is for libraries.

**Comment by neheb** [comment]
I think I found the issue. LTO bug.

**Comment by neheb** [comment]
Add this patch:

```
--- a/meson.build
+++ b/meson.build
@@ -7,7 +7,6 @@ project(
     version: '5.0.1',
     license: 'LGPLv2+',
     default_options: [
-        'b_lto=true',
         'b_lto_mode=thin',
         'b_colorout=always',
         'b_asneeded=true',
```

and

```
--- a/utils/lxc/Makefile
+++ b/utils/lxc/Makefile
@@ -25,6 +25,7 @@ include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/meson.mk

 MESON_ARGS += \
+       -Db_lto=$(if $(findstring mips,$(CONFIG_ARCH)),false,true) \
        -Dwerror=false \
        -Dinit-script=sysvinit \
        -Dman=false \
```

**Comment by eli-schwartz** [comment]
If you're overriding the setting anyway, you don't care what its project default_options is.

**Comment by neheb** [comment]
oh ok. Edited.

**Comment by graysky2** [comment]
I applied @neheb's patch and also added back `-Db_pie` which I removed as a test.  Let's see how the checks go.

File: utils/lxc/Makefile [hunk 3]
```diff
@@ -9,22 +9,31 @@
...
+MESON_ARGS += \
+	-Db_lto=$(if $(findstring mips,$(CONFIG_ARCH)),false,true) \
+	-Dwerror=false \
+	-Dinit-script=sysvinit \
+	-Dman=false \
+	-Dapparmor=false \
+	-Dselinux=false \
+	-Dseccomp=$(if $(CONFIG_LXC_SECCOMP),true,false) \
+	-Dexamples=false \
+	-Db_pie
```
Comment (neheb) [inline utils/lxc/Makefile:36]: =true. I think this is already default though.

**Comment by graysky2** [comment]
That worked... trying now with the `-Db_pie=true` removed since it could be defaulted and therefore superfluous.

**Comment by AreYouLoco** [comment]
So this can be merged now?

**Comment by graysky2** [comment]
Pull it: `git pull origin pull/18990/head --no-edit` or put the word `.patch` in your browser's url to get a patch

**Comment by AreYouLoco** [comment]
Ok thanks @graysky2 I got the patch.

**Comment by AreYouLoco** [comment]
Could it be somehow backported to 21.02?

**Comment by neheb** [comment]
21.02 needs meson/host as tools/meson was not around.

**Comment by AreYouLoco** [comment]
@neheb could you point me into direction on how to build it for 21.02?

**Comment by graysky2** [comment]
Do you have a good reason to stay with the LTS release?  Building from master has been rock solid for me for a long time now.

**Comment by AreYouLoco** [comment]
Actually I am running OpenWRT fork that is based on 21.02 and I could move to developer branches based on master but I am a bit worried about stability

**Comment by neheb** [comment]
@AreYouLoco 

```
make package/meson/host/compile
make package/lxc/compile
```

should work.

**Comment by AreYouLoco** [comment]
I started with clean sources. Added a patch from here to update lxc to 5.0.1 and the first command finished but the second complained about "no rule to make target [...]/meson.mk" Is this the right place to ask for backport?

**Comment by neheb** [comment]
Oh right. The meson.mk line needs to be changed. See https://github.com/openwrt/packages/blob/openwrt-21.02/sound/mpd/Makefile#L29

**Comment by AreYouLoco** [comment]
```
ERROR: Meson version is 0.56.0 but project requires >= 0.61
```

So I guess I need to port meson somehow

**Comment by graysky2** [comment]
Not sure how involved that will be (does meson have deps) but do that under your openwrt build dir not under `feeds/packages`

**Comment by neheb** [comment]
meson in 21.02 lives in devel/meson so update that.

**Comment by AreYouLoco** [comment]
I tried but my changes were overwritten and I got the same error. How do I do that?

**Pull request merged**